src/meson.build

```diff
@@ -8,6 +8,7 @@ add_project_arguments('-Wall', language : 'c')
 add_project_arguments('-Werror', language : 'c')
 add_project_arguments('-fvisibility=hidden', language : 'c')
 add_project_arguments('-DWL_HIDE_DEPRECATED', language : 'c')
+add_project_arguments('-D_GNU_SOURCE', language : 'c')
```

Comment:
It's not clear to me autotools always enables _GNU_SOURCE. It might be getting enabled as a side effect of something else, but not reliably. A little Googling found this, for example:

https://reversed.top/2013-10-16/autoconf-how-to-define-gnu-source-correctly/

Reply:
Ah, yeah, that's a good point. I had based that on some local testing, but if there's an autoconf macro for that, then we should use it.